test/helpers/nodestate.go

```diff
@@ -0,0 +1,71 @@
+package helpers
+
+import (
+	"github.com/openshift/machine-config-operator/pkg/daemon/constants"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Determines whether a node has changed configs.
+func hasNodeConfigChanged(node corev1.Node, mcName string) bool {
+	current := node.Annotations[constants.CurrentMachineConfigAnnotationKey]
+	desired := node.Annotations[constants.DesiredMachineConfigAnnotationKey]
+
+	return current == desired && desired == mcName
+}
+
+// Determines whether a node has changed images.
+func hasNodeImageChanged(node corev1.Node, image string) bool {
+	current := node.Annotations[constants.CurrentImageAnnotationKey]
+	desired := node.Annotations[constants.DesiredImageAnnotationKey]
+
+	return current == desired && desired == image
+}
+
+// Determines whether the MCD on a given node has reached the "Done" state.
+func isMCDDone(node corev1.Node) bool {
+	state := node.Annotations[constants.MachineConfigDaemonStateAnnotationKey]
+	return state == constants.MachineConfigDaemonStateDone
+}
+
+// Determines if a given node is ready.
+func isNodeReady(node corev1.Node) bool {
```

Comment:
IMO this works for me in the test context, but should we be concerned about the [other variation](https://github.com/openshift/machine-config-operator/blob/005ee68664fa12db7ec159deb984b30421331950/pkg/controller/common/layered_node_state.go#L309) of this function we have in production code? Should they be identical?

Reply:
That's a great point. I think I might add a variation of these helpers onto the LayeredNodeState object and use that to interrogate the node objects instead. It would be great from a consistency perspective.

Reply:
No problem, that works for me 🙂

As a sidenote https://github.com/openshift/machine-config-operator/pull/4489 also works in that area and while it is not doing a complete refactor, maybe that can be a starting point?